packages/angular/projects/ui-angular/src/lib/primitives-poc/icon/icon.component/icon.component.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  ViewEncapsulation,
+} from '@angular/core';
+import { AmplifyBasePrimitiveComponent } from '../../base-primitive/components/base-primitive.component';
+
+const $COMPONENT_SELECTOR: string = 'amplify-icon';
+@Component({
+  selector: $COMPONENT_SELECTOR,
+  templateUrl: 'icon.component.html',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AmplifyIconComponent extends AmplifyBasePrimitiveComponent {
+  /** Input Prop for style changes  */
+
+  @Input() pathData: string;
+
+  public _viewBox: string = '';
+  @Input()
+  public set viewBox(viewBox: { width: string; height: string }) {
```

Comment:
`width` and `height` should be `number` and not `string` ([ref](https://ui.docs.amplify.aws/react/components/icon)).

Reply:
Can we define this type somewhere outside `AmplifyIconComponent` for separation of concern? 

e.g.

```ts
// somewhere above this file:
interface ViewBox {
  width: number;
  height: number;
}

// inside `AmplifyIconComponent`:
{
  // ...
  @Input()
  public set viewBox(viewBox: ViewBox) 
  // ...
}
```

Reply:
[28467b5](https://github.com/aws-amplify/amplify-ui/pull/2436/commits/28467b5846cc0bb40ac862d24bcca96b0e532b1a)